.github/workflows/sync-examples.yml

```diff
@@ -31,16 +34,25 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 2 # fetch 2 to compare with previous commit for changes
+          ref: ${{ inputs.checkout-ref }}
 
       - name: Detect changesets
         uses: bluwy/detect-changesets-action@v1
         id: detect
 
+      - name: Get pre mode of changesets
+        id: pre
+        run: |
+          if [ -f ./.changeset/pre.json ]; then
+            pre_value=$(jq -r '.tag' ./.changeset/pre.json)
+            echo "value=$pre_value" >> $GITHUB_OUTPUT
+          fi
+
       # We only do sync if there are no changesets, so we don't accidentally allow users
       # to clone examples that may rely on unreleased code
 
-      - name: Sync from main branch to latest and examples/* branches
-        if: steps.detect.outputs.has-changesets == 'false' && github.ref == 'refs/heads/main'
+      - name: Sync stable to latest and examples/* branches
+        if: steps.detect.outputs.has-changesets == 'false' && github.ref == 'refs/heads/main' && steps.pre.outputs.value == ''
```

Comment:
We technically no longer need the `github.ref == 'refs/heads/main'` part, but I think it's better to leave that in to prevent cases where we create the `next` branch again but haven't entered prerelease, and we don't want to accidentally sync it to the `latest` branch examples.